MetaLamp/nft-marketplace/src/Plutus/Contracts/NftMarketplace/OnChain/Core/NFT.hs

```diff
@@ -42,16 +42,40 @@ import qualified PlutusTx.AssocMap                      as AssocMap
 import           PlutusTx.Prelude                       hiding (Semigroup (..))
 import           Prelude                                (Semigroup (..))
 import qualified Prelude                                as Haskell
+import           Control.Monad          (join)
 
 -- Category = [BuiltinByteString]
 -- 1. acts as a list of category with nested subcategories
 -- 2. acts as a list of tags
 type IpfsCid = BuiltinByteString
 type IpfsCidHash = BuiltinByteString
 type Category = [BuiltinByteString]
-type LotLink = Either Sale.Sale Auction.Auction
 type BundleId = BuiltinByteString
 
+data LotLink = 
+  SaleLotLink Sale.Sale 
+  | AuctionLotLink Auction.Auction
+  deriving stock (Haskell.Eq, Haskell.Show, Haskell.Generic)
+  deriving anyclass (J.ToJSON, J.FromJSON)
+
+PlutusTx.unstableMakeIsData ''LotLink
+
+PlutusTx.makeLift ''LotLink
+
+Lens.makeClassy_ ''LotLink
+
+getSaleFromLot :: LotLink -> Maybe Sale.Sale
```

Comment:
No need for these two functions(`getAuctionFromLot`, `getAuctionFromLot`) - we can use `^?` instead.

Reply:
Good point, that operator works)